### DISCUSSION TITLE
Get the true current route rather than screen's route?

### DISCUSSION THREAD

**Discussion by traviswimer** [react-navigation/react-navigation#11074] [Q&A] [answered]
I have a situation where I need to access the current `route` from outside the current screen.

In other words, I need the functionality of `NavigationContainer`'s [getCurrentRoute()](https://reactnavigation.org/docs/navigation-container/#getcurrentroute) method, but within a screen.

The [useRoute()](https://reactnavigation.org/docs/use-route/) hook doesn't work, because it only returns the route for the screen it is called within. If `useRoute` is in a high level screen and you `navigate()` to a deeply-nested screen, you wont receive the route information for that deeply-nested screen.

Is there any built in way to do this, or will I need to create my own implementation using `NavigationContainer`'s `getCurrentRoute()`?

**Comment by satya164** [comment, marked as answer]
There's no built-in API to do it. But you can implement such a hook using the `ref` and adding the `state` listener on it.

Keep in mind that this means that the screen is going to re-render on every navigation, potentially slowing down the other navigations - unless you only need that info in an event listener and can skip the re-render.
